jenkins/e2e-image/e2e-runner.sh

```diff
@@ -411,7 +411,7 @@ if [[ -n "${BOOTSTRAP_MIGRATION:-}" ]]; then
   # until that time. This environment variable will do that.
   source "$(dirname "${0}")/upload-to-gcs.sh"
   version=$(find_version)  # required by print_started
-  print_started | jq '.metadata?' > "${ARTIFACTS}/metadata.json"
+  print_started | jq '.metadata? + {version, "job-version"}' > "${ARTIFACTS}/metadata.json"
```

Comment:
I agree that this is getting really hard to follow, and it seems pretty fragile.

a comment would be good.

it'd also be nice at some point to have some canonical understanding of what's expected in `started.json`, `metadata.json`, and `finished.json`. it seems to be loosely specified in many different places, which clearly do not all line up.